### PR TITLE
Soluion

### PR DESCRIPTION
**Pull request by SashaViun**

### CODE REVIEW

**Review by NVyrsta — APPROVED** [review]
Good job :)

File: src/sortStudents.ts [hunk 1]
```diff
@@ -1,16 +1,54 @@
...
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+const getAverage = (marks:number[]): number => {
+  const averageMark:number = marks
+    .reduce((prevNum: number, nextNum: number): number => prevNum + nextNum, 0)
+    / marks.length;
```
Comment (ArturErmolenko) [inline src/sortStudents.ts:24]:
```suggestion
 return marks
    .reduce((acc, curr)=> acc + curr, 0)
    / marks.length;
```

File: src/sortStudents.ts [hunk 2]
```diff
@@ -1,16 +1,54 @@
...
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return averageMark;
```
Comment (ArturErmolenko) [inline src/sortStudents.ts:26]:
```suggestion
```

File: src/sortStudents.ts (hunk 1 above)
Comment (ArturErmolenko) [reply src/sortStudents.ts:24]: you can return immediately

File: src/sortStudents.ts [hunk 3]
```diff
@@ -1,16 +1,54 @@
...
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+const getAverage = (marks:number[]): number => {
```
Comment (ArturErmolenko) [inline src/sortStudents.ts:21]:
```suggestion
const getAverageGrades = (marks:number[]): number => {
```

File: src/sortStudents.ts [hunk 4]
```diff
@@ -1,16 +1,54 @@
...
+      case SortType.Married:
+        return order === 'asc'
+          ? +a[sortBy] - +b[sortBy]
+          : +b[sortBy] - +a[sortBy];
+
+      case SortType.AverageGrade:
+        return order === 'asc'
+          ? getAverage(a[sortBy]) - getAverage(b[sortBy])
+          : getAverage(b[sortBy]) - getAverage(a[sortBy]);
+      default: throw Error('Parameters are not correct');
```
Comment (ArturErmolenko) [inline src/sortStudents.ts:51]: add empty lines between cases
```suggestion
  return [...students].sort((a, b) => {
    switch (sortBy) {
      case SortType.Name:
      case SortType.Surname:
        return order === 'asc'
          ? a[sortBy].localeCompare(b[sortBy])
          : b[sortBy].localeCompare(a[sortBy]);
          
      case SortType.Age:
      case SortType.Married:
        return order === 'asc'
          ? +a[sortBy] - +b[sortBy]
          : +b[sortBy] - +a[sortBy];

      case SortType.AverageGrade:
        return order === 'asc'
          ? getAverage(a[sortBy]) - getAverage(b[sortBy])
          : getAverage(b[sortBy]) - getAverage(a[sortBy]);
          
      default: throw Error('Parameters are not correct');
```

**Review by ArturErmolenko — CHANGES_REQUESTED** [review]
Well done 👍 
Left comments ⭐

File: src/sortStudents.ts [hunk 5]
```diff
@@ -1,16 +1,69 @@
...
+};
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+):
+  Student[] {
```
Comment (alexandra-protyanova) [inline src/sortStudents.ts:32]:
```suggestion
export function sortStudents(
  students: Student[],
  sortBy: SortType,
  order: SortOrder,
): Student[] {
```

**Review by alexandra-protyanova — APPROVED** [review]
Well done 🔥

**Pull request closed without merging**